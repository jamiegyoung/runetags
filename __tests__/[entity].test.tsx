import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import Entity, { getStaticPaths, getStaticProps } from '@/pages/[entity]';

const TestEntity = (
  alt = false,
  guide = false,
  source = false,
  modified = false,
  subcategory = false,
) => (
  <Entity
    safeURI={`AbyssalSire`}
    name={`Abyssal Sire`}
    subcategory={subcategory ? `boss` : undefined}
    altName={alt ? `Abyssal Sire Alt Test` : undefined}
    source={
      source
        ? {
            name: `AsukaYen OSRS - OSRS Abyssal Sire Guide [2021]`,
            link: `https://www.youtube.com/watch?v=wnZJl9driUs`,
            modified: modified ? `true` : undefined,
          }
        : undefined
    }
    tags={[`slayer`, `boss`]}
    recommendedGuideVideoId={guide ? `wnZJl9driUs` : undefined}
    items={[
      {
        regionId: 11850,
        regionX: 25,
        regionY: 34,
        z: 0,
        color: `#FFFFFF00`,
      },
    ]}
    thumbnail={`https://oldschool.runescape.wiki/images/thumb/Abyssal_Sire.png/200px-Abyssal_Sire.png`}
    wiki={`https://oldschool.runescape.wiki/w/Abyssal_Sire`}
    fullName={`Abyssal Sire`}
    fullAltName={alt ? `Abyssal Sire Alt Test` : ``}
  />
);

describe(`Entity`, () => {
  it(`should render correctly`, () => {
    const { container } = render(TestEntity());
    expect(container).toMatchSnapshot();
  });

  it(`should render correctly with a subcategory`, () => {
    const { getByText } = render(TestEntity(false, false, false, false, true));
    expect(getByText(`(boss)`)).toBeInTheDocument();
  });

  it(`should render an optional alt name`, () => {
    const { getByText } = render(TestEntity(true));
    expect(getByText(`Abyssal Sire Alt Test`)).toBeInTheDocument();
  });

  it(`should render an optional guide`, () => {
    const { getByText } = render(TestEntity(false, true));
    expect(getByText(`Recommended Guide`)).toBeInTheDocument();
  });

  it(`should render an optional source`, () => {
    const { getByText } = render(TestEntity(false, false, true));
    expect(getByText(`Source`)).toBeInTheDocument();
  });

  it(`should generate the correct RuneLite tiles link`, () => {
    const { getByText } = render(TestEntity());
    const link = getByText(`View Map on RuneLite`);
    link.hasAttribute(`href`);

    expect(link.getAttribute(`href`)).toBe(
      `https://runelite.net/tile/show/#W3sicmVnaW9uSWQiOjExODUwLCJyZWdpb25YIjoyNSwicmVnaW9uWSI6MzQsInoiOjAsImNvbG9yIjoiI0ZGRkZGRjAwIn1d`,
    );
  });

  it(`should generate the modified text if the source has been modified`, () => {
    const { getByText } = render(TestEntity(false, false, true, true));
    expect(getByText(`(modified)`)).toBeInTheDocument();
  });

  it(`should get the correct static props`, async () => {
    const { props } = await getStaticProps({
      params: { entity: `abyssal sire` },
    });

    expect(props?.name).toBe(`Abyssal Sire`);
    expect(props?.tags).toEqual([`slayer`, `boss`]);
  });

  it(`should get the correct static paths`, async () => {
    const { paths } = await getStaticPaths();
    expect(paths).toContainEqual({
      params: { entity: `abyssal sire` },
    });
  });
});
