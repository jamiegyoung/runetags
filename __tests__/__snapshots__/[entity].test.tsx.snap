// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entity should render correctly 1`] = `
<div>
  <div
    class="navbar"
  >
    <a
      href="./"
      style="background: transparent; color: rgb(245, 243, 245); text-decoration: none; display: flex;"
    >
      <header
        class="title"
        style="font-size: 2rem;"
      >
        R
        <span>
          une
        </span>
        <span
          class="titleAlt"
        >
          M
          <span>
            arkers
          </span>
        </span>
      </header>
    </a>
    <a
      class="navbarLink"
      href="/tags"
    >
      Rune
      <span>
        Tags
      </span>
    </a>
  </div>
  <div
    class="container"
  >
    <div
      class="innerListContainer"
    >
      <div
        class="card"
      >
        <a
          class="imageContainer"
          href="/AbyssalSire"
        >
          <img
            alt="Abyssal Sire image"
            class="image"
            data-nimg="1"
            decoding="async"
            height="100"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Foldschool.runescape.wiki%2Fimages%2Fthumb%2FAbyssal_Sire.png%2F200px-Abyssal_Sire.png&w=256&q=75"
            srcset="/_next/image?url=https%3A%2F%2Foldschool.runescape.wiki%2Fimages%2Fthumb%2FAbyssal_Sire.png%2F200px-Abyssal_Sire.png&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Foldschool.runescape.wiki%2Fimages%2Fthumb%2FAbyssal_Sire.png%2F200px-Abyssal_Sire.png&w=256&q=75 2x"
            style="color: transparent; opacity: 0;"
            width="100"
          />
        </a>
        <div
          class="cardData"
        >
          <div
            class="entityInfo"
          >
            <a
              class="name"
              href="/AbyssalSire"
            >
              Abyssal Sire
            </a>
          </div>
          <div>
            <p
              class="tileCount"
            >
              1
               tile markers
            </p>
            <div
              class="tileInteraction"
            >
              <button
                class="button button"
              >
                Copy
                <div
                  class="animBar "
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="linkContainer"
      >
        <a
          class="link"
          href="https://runelite.net/tile/show/#W3sicmVnaW9uSWQiOjExODUwLCJyZWdpb25YIjoyNSwicmVnaW9uWSI6MzQsInoiOjAsImNvbG9yIjoiI0ZGRkZGRjAwIn1d"
          rel="noopener noreferrer"
          target="_blank"
        >
          View Map on RuneLite
        </a>
        <a
          class="link"
          href="https://oldschool.runescape.wiki/w/Abyssal_Sire"
          rel="noopener noreferrer"
          target="_blank"
        >
          Wiki Page
        </a>
      </div>
      <div
        style="display: flex; flex-direction: column; align-items: center; width: 90%; margin: 10px 0px 10px 0px;"
      >
        <h2
          style="margin: 10px 10px 20px 10px; align-self: flex-start;"
        >
          Tile Data
        </h2>
        <div
          class="options"
        >
          <code
            aria-pressed="false"
            class="togglePretty "
            role="button"
          >
            pretty print
          </code>
        </div>
        <code
          class="code"
          style="white-space: normal;"
        >
          [{"regionId":11850,"regionX":25,"regionY":34,"z":0,"color":"#FFFFFF00"}]
        </code>
      </div>
    </div>
  </div>
  <footer
    class="footer"
  >
    <div
      class="footerItem"
    >
      <p>
        Made by
        <a
          class="link"
          href="https://discord.com/users/167850724976361472"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="GitHub"
            data-nimg="1"
            decoding="async"
            height="17"
            loading="lazy"
            src="/_next/image?url=%2Fdiscord.png&w=48&q=75"
            srcset="/_next/image?url=%2Fdiscord.png&w=32&q=75 1x, /_next/image?url=%2Fdiscord.png&w=48&q=75 2x"
            style="color: transparent;"
            width="20"
          />
          <span
            class="linkText"
          >
            Jam#0001
          </span>
        </a>
      </p>
    </div>
    <div
      class="footerItem"
    >
      <a
        class="link"
        href="https://github.com/jamiegyoung/runemarkers"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="GitHub"
          data-nimg="1"
          decoding="async"
          height="20"
          loading="lazy"
          src="/_next/image?url=%2Fgithub.png&w=48&q=75"
          srcset="/_next/image?url=%2Fgithub.png&w=32&q=75 1x, /_next/image?url=%2Fgithub.png&w=48&q=75 2x"
          style="color: transparent;"
          width="20"
        />
        <span
          class="linkText"
        >
          contribute code / tiles
        </span>
      </a>
    </div>
  </footer>
</div>
`;
